src/sass/_theme_rst.sass

```diff
@@ -223,9 +223,10 @@
   // Sphinx can highlight searched text with ?highlighted=searchterm
   .highlighted
     background: $highlight-color
-    display: inline-block
+    box-shadow: 0 0 0 3px $highlight-color
```

Comment:
If you change 3px into 2px, the example you gave will not have the highlights stick together.